ext/curl/interface.c

```diff
@@ -2529,34 +2536,64 @@ static int _php_curl_setopt(php_curl *ch, zend_long option, zval *zvalue) /* {{{
 						zval *prop, rv;
 						char *type = NULL, *filename = NULL;
 
-						prop = zend_read_property(curl_CURLFile_class, current, "name", sizeof("name")-1, 0, &rv);
-						if (Z_TYPE_P(prop) != IS_STRING) {
-							php_error_docref(NULL, E_WARNING, "Invalid filename for key %s", ZSTR_VAL(string_key));
-						} else {
-							postval = Z_STR_P(prop);
+						prop = zend_read_property(curl_CURLFile_class, current, "mime", sizeof("mime")-1, 0, &rv);
+						if (Z_TYPE_P(prop) == IS_STRING && Z_STRLEN_P(prop) > 0) {
+							type = Z_STRVAL_P(prop);
+						}
 
-							if (php_check_open_basedir(ZSTR_VAL(postval))) {
-								return 1;
-							}
+						prop = zend_read_property(curl_CURLFile_class, current, "postname", sizeof("postname")-1, 0, &rv);
+						if (Z_TYPE_P(prop) == IS_STRING && Z_STRLEN_P(prop) > 0) {
+							filename = Z_STRVAL_P(prop);
+						}
 
-							prop = zend_read_property(curl_CURLFile_class, current, "mime", sizeof("mime")-1, 0, &rv);
-							if (Z_TYPE_P(prop) == IS_STRING && Z_STRLEN_P(prop) > 0) {
-								type = Z_STRVAL_P(prop);
-							}
-							prop = zend_read_property(curl_CURLFile_class, current, "postname", sizeof("postname")-1, 0, &rv);
-							if (Z_TYPE_P(prop) == IS_STRING && Z_STRLEN_P(prop) > 0) {
-								filename = Z_STRVAL_P(prop);
+						prop = zend_read_property(curl_CURLFile_class, current, "name", sizeof("name")-1, 0, &rv);
+						if (zval_is_true(prop)) {
```

Comment:
Not sure it can be UNDEF, but I'd rather specify explicitly which values we do expect - i.e. if we expect _name_ to be NULL when we use _buffer_.

Also, whatever check you do here you should also do on __wakeup. In fact, I'd make it a macro or inline function to ensure they are always the same.